enterprise/server/mockoauth/mockoauth.go

```diff
@@ -0,0 +1,253 @@
+			Issuer:                           o.IssuerURL().String(),
+			AuthorizationEndpoint:            o.AuthorizationEndpoint().String(),
+			TokenEndpoint:                    o.TokenEndpoint().String(),
+			JwksUri:                          o.JwksEndpoint().String(),
+			IdTokenSigningAlgValuesSupported: []string{"RS256"},
+		},
+	)
+}
+
+func (o *oauth2Mock) IssuerURL() *url.URL {
+	return o.env.GetConfigurator().GetMockOauthIssuer()
+}
+
+func (o *oauth2Mock) AuthorizationEndpoint() *url.URL {
+	u := o.IssuerURL()
+	u.Path = "/login/oauth/authorize"
+	return u
+}
+
+func (o *oauth2Mock) TokenEndpoint() *url.URL {
+	u := o.IssuerURL()
+	u.Path = "/login/oauth/access_token"
+	return u
+}
+
+func (o *oauth2Mock) JwksEndpoint() *url.URL {
+	u := o.IssuerURL()
+	u.Path = "/.well-known/jwks.json"
+	return u
+}
+
+// AuthCodeURL redirects to our own server.
+func (o *oauth2Mock) AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string {
+	u := o.AuthorizationEndpoint()
+
+	v := url.Values{}
+	v.Set("state", state)
+
+	u.RawQuery = v.Encode()
+	return u.String()
+}
+
+// OAuthAuthorize handles requests to /login/oauth/authorize.
+func (o *oauth2Mock) Authorize(w http.ResponseWriter, r *http.Request) {
+	state := r.FormValue("state")
+
+	u := o.env.GetConfigurator().GetMockOauthIssuer()
+	u.Path = "/auth/"
+
+	v := url.Values{}
+	v.Set("code", "code")
+	v.Set("state", state)
+	u.RawQuery = v.Encode()
+
+	http.Redirect(w, r, u.String(), http.StatusTemporaryRedirect)
+
+}
+
+func writeJSONResponse(w http.ResponseWriter, r *http.Request, v interface{}) {
+	rsp, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json;charset=UTF-8")
+	w.Write(rsp)
+}
+
+// AccessToken handles requests to /login/oauth/access_token, taking the code and returning a real token.
+func (o *oauth2Mock) AccessToken(w http.ResponseWriter, r *http.Request) {
+	idKey := make([]byte, 32)
+	_, err := rand.Read(idKey)
+	if err != nil {
+		log.Errorf("Error reading rand: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	token := jwt.New()
+	token.Set(jwt.AudienceKey, "buildbuddy")
+	token.Set(jwt.ExpirationKey, time.Now().Add(time.Hour).Unix())
+	token.Set(jwt.JwtIDKey, base64.StdEncoding.EncodeToString(idKey))
+	token.Set(jwt.IssuedAtKey, time.Now().Unix())
+	token.Set(jwt.IssuerKey, o.IssuerURL().String())
+	token.Set(jwt.SubjectKey, "")
+
+	pictureUrl := o.IssuerURL()
+	pictureUrl.Path = "/favicon/favicon_black.svg"
+	token.Set("email", "buildbuddy@example.com")
+	token.Set("name", "buildbuddy")
+	token.Set("given_name", "Build")
+	token.Set("family_name", "Buddy")
+	token.Set("picture", pictureUrl.String())
+
+	signed, err := jwt.Sign(token, jwa.RS256, o.rsaPrivateKey)
+	if err != nil {
+		log.Errorf("Error signing token: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	writeJSONResponse(
+		w,
+		r,
+		&tokenJSON{
```

Comment:
I think a more idiomatic formatting of these would be: writeJSONResponse(w, r, &tokenJSON{
....})